apps/condo/domains/notification/utils/serverSchema/messageTypeHooks/AbstractHooksSet.js

```diff
@@ -0,0 +1,30 @@
+const { noop } = require('lodash')
+
+/**
+ * @abstract
+ */
+class AbstractHooksSet {
```

Comment:
it's redundent abstraction level. It's better yo use it inside the code

Reply:
I'd like to define a strict interface for message lifecycle events. If some message type needs to use only one hook (event) the other ones already have a default value that will not crash the delivery process.
So, to add hooks for some message type do the following:
1. Extend abstract class and realize needed functions
2. Add class to MESSAGE_META

Reply:
Refactored